### PR TITLE
Adding extra error handling as already exists error occurs

### LINKED ISSUE #1939: [BUG]: Existing catalog and schemas should be skipped in the cli command create-catalogs-schemas
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

create-catalogs-schemas fails with existing catalog or schema error.

### Expected Behavior

Skip existing catalogs and schemas if they are already created.

### Steps To Reproduce

_No response_

### Cloud

AWS

### Operating System

Linux

### Version

latest via Databricks CLI

### Relevant log output

```shell
1:49:21 ERROR [d.l.ucx.create-catalogs-schemas] Failed to call create-catalogs-schemas: Traceback (most recent call last):

  File "/home/<username redacted>/.databricks/labs/ucx/state/venv/lib/python3.10/site-packages/databricks/labs/blueprint/cli.py", line 105, in _route

    cmd.fn(**kwargs)

  File "/home/<username redacted>/.databricks/labs/ucx/lib/src/databricks/labs/ucx/cli.py", line 357, in create_catalogs_schemas

    ctx.catalog_schema.create_all_catalogs_schemas(prompts)

  File "/home/<username redacted>/.databricks/labs/ucx/lib/src/databricks/labs/ucx/hive_metastore/catalog_schema.py", line 32, in create_all_catalogs_schemas

    self._create_catalog_validate(candidate_catalog, prompts)

  File "/home/<username redacted>/.databricks/labs/ucx/lib/src/databricks/labs/ucx/hive_metastore/catalog_schema.py", line 92, in _create_catalog_validate

    self._create_catalog(catalog, catalog_storage)

  File "/home/<username redacted>/.databricks/labs/ucx/lib/src/databricks/labs/ucx/hive_metastore/catalog_schema.py", line 160, in _create_catalog

    self._ws.catalogs.create(catalog, comment="Created by UCX")

  File "/home/<username redacted>/.databricks/labs/ucx/state/venv/lib/python3.10/site-packages/databricks/sdk/service/catalog.py", line 6176, in create

    res = self._api.do('POST', '/api/2.1/unity-catalog/catalogs', body=body, headers=headers)

  File "/home/<username redacted>/.databricks/labs/ucx/state/venv/lib/python3.10/site-packages/databricks/sdk/core.py", line 130, in do

    response = retryable(self._perform)(method,

  File "/home/<username redacted>/.databricks/labs/ucx/state/venv/lib/python3.10/site-packages/databricks/sdk/retries.py", line 54, in wrapper

    raise err

  File "/home/<username redacted>/.databricks/labs/ucx/state/venv/lib/python3.10/site-packages/databricks/sdk/retries.py", line 33, in wrapper

    return func(*args, **kwargs)

  File "/home/<username redacted>/.databricks/labs/ucx/state/venv/lib/python3.10/site-packages/databricks/sdk/core.py", line 238, in _perform

    raise self._make_nicer_error(response=response, **payload) from None

databricks.sdk.errors.platform.BadRequest: Catalog 'dev_catalog' already exists
```

### PR DESCRIPTION
**Pull request by pritishpai**
## Changes
CLI command create-catalogs-schemas sometimes fails if the catalog/schema already exists. The _get_missing_catalogs_schemas should help to avoid re-creating existing ones, but this seems like an edge case.

### Linked issues
Resolves #1939 

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [x] manually tested
- [ ] added unit tests
- [ ] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

**Comment by pritishpai** [comment]
image

File: src/databricks/labs/ucx/hive_metastore/catalog_schema.py [hunk 1]
```diff
@@ -29,10 +30,22 @@ def __init__(
     def create_all_catalogs_schemas(self, prompts: Prompts):
         candidate_catalogs, candidate_schemas = self._get_missing_catalogs_schemas()
         for candidate_catalog in candidate_catalogs:
-            self._create_catalog_validate(candidate_catalog, prompts)
+            try:
+                self._create_catalog_validate(candidate_catalog, prompts)
+            except BadRequest as e:
+                if "already exists" in str(e):
```
Comment (nfx) [inline src/databricks/labs/ucx/hive_metastore/catalog_schema.py:36]: is it really BadRequest?

image

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/hive_metastore/catalog_schema.py (hunk 1 above)
Comment (pritishpai) [reply src/databricks/labs/ucx/hive_metastore/catalog_schema.py:36]: This is the error that we are getting

`databricks.sdk.errors.platform.BadRequest: Catalog 'dev_catalog' already exists`

Comment (JCZuurmond) [inline src/databricks/labs/ucx/hive_metastore/catalog_schema.py:36]: Maybe move this into a (static)method `self._error_already_exists(e)`

File: tests/unit/test_cli.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 from databricks.sdk.service.catalog import ExternalLocationInfo
 from databricks.sdk.service.compute import ClusterDetails, ClusterSource
 from databricks.sdk.service.workspace import ObjectInfo, ObjectType
+from databricks.sdk.errors.platform import BadRequest
```
Comment (JCZuurmond) [inline tests/unit/test_cli.py:16]: isort

File: tests/unit/test_cli.py [hunk 3]
```diff
@@ -432,6 +433,18 @@ def test_create_catalogs_schemas(ws):
     ws.catalogs.list.assert_called_once()
 
 
+def test_create_catalogs_schemas_handles_existing(ws, caplog):
+    prompts = MockPrompts({'.*': 's3://test'})
```
Comment (JCZuurmond) [inline tests/unit/test_cli.py:437]: Should the answer to all prompts be `"s3://test"`?

**Review by JCZuurmond — COMMENTED** [review]
Added some nits, generally looks good

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**